File: util/talk/web/src/lib/internal-evidence.js
// App reference records are not answer text. Keep literal code examples intact.
export function stripInternalEvidence(value) {
  let fence = '';
  const marker = '[Historical tool evidence:';
  return String(value || '').split('\n').filter(line => {
    const trimmed = line.trim();
    if (/^(?:```|~~~)/.test(trimmed)) {
      const next = trimmed.slice(0, 3);
      if (!fence) fence = next; else if (fence === next) fence = '';
      return true;
    }
    if (fence) return true;
    if (/^\[Historical tool evidence: [\w.-]+; archive_id=\d+; use context_read for original\]/.test(trimmed)) return false;
    // Suppress a marker while its streamed line is still incomplete.
    if (trimmed.length >= 11 && marker.startsWith(trimmed)) return false;
    if (trimmed.startsWith(marker) && !trimmed.includes(']')) return false;
    return true;
  }).join('\n');
}

File: util/talk/internal/server/internal_evidence.go
package server

import (
	"regexp"
	"strings"
)

var historicalEvidenceLine = regexp.MustCompile(`^\[Historical tool evidence: [\w.-]+; archive_id=\d+; use context_read for original\]`)

// Strip only app-generated reference records outside literal code examples.
func cleanInternalEvidence(text string) string {
	lines := strings.Split(text, "\n")
	out := make([]string, 0, len(lines))
	fence := ""
	for _, line := range lines {
		trimmed := strings.TrimSpace(line)
		if strings.HasPrefix(trimmed, "```") || strings.HasPrefix(trimmed, "~~~") {
			marker := trimmed[:3]
			if fence == "" {
				fence = marker
			} else if marker == fence {
				fence = ""
			}
			out = append(out, line)
			continue
		}
		if fence == "" && historicalEvidenceLine.MatchString(trimmed) {
			continue
		}
		out = append(out, line)
	}
	return strings.TrimSpace(strings.Join(out, "\n"))
}
